### DISCUSSION TITLE
v3.1.0

### DISCUSSION THREAD

**Discussion by zpao** [zpao/qrcode.react#218] [Announcements]
### Fixed
- Made optional props optional, removing use of `defaultProps`. This may be a slight behavior change with TypeScript if previously passing `null`.
- Fixed used of `shapeRendering="crispEdges"` in SVG component.

<hr /><em>This discussion was created from the release <a href='https://github.com/zpao/qrcode.react/releases/tag/v3.1.0'>v3.1.0</a>.</em>

**Comment by luan-hte** [comment]
How can I make the size responsive, with tailwind css for example?

**Reply by zpao** [reply]
You can specify styles or class names to pass to the svg/canvas elements directly. You can do something like `<QRCode style={{width: '100%'}} ... />`

**Comment by blackfy** [comment]
How to change positioning scale to round or eye shape

**Reply by zpao** [reply]
That's not supported at this time. https://github.com/zpao/qrcode.react/issues/124 has some discussion.

**Comment by seansystem6** [comment]
Hi @zpao is there a way that we could convert the canvas to an image?
